Gems/ROS2/Code/Source/Spawner/ROS2SpawnerEditorComponent.cpp

```diff
@@ -0,0 +1,102 @@
+        if (serializeContext)
+        {
+            serializeContext->Class<ROS2SpawnerEditorComponent, ROS2SpawnerEditorComponentBase>()->Version(1);
+
+            AZ::EditContext* editContext = serializeContext->GetEditContext();
+            if (editContext)
+            {
+                editContext->Class<ROS2SpawnerEditorComponent>("ROS2 Spawner", "Spawner component")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "Manages spawning of robots in configurable locations")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly);
+            }
+        }
+    }
+
+    AZStd::unordered_map<AZStd::string, SpawnPointInfo> ROS2SpawnerEditorComponent::GetSpawnPoints() const
+    {
+        AZStd::vector<AZ::EntityId> children;
+        AZ::TransformBus::EventResult(children, m_controller.m_config.m_editorEntityId, &AZ::TransformBus::Events::GetChildren);
+
+        AZStd::unordered_map<AZStd::string, SpawnPointInfo> result;
+
+        for (const AZ::EntityId& child : children)
+        {
+            AZ::Entity* childEntity = nullptr;
+            AZ::ComponentApplicationBus::BroadcastResult(childEntity, &AZ::ComponentApplicationRequests::FindEntity, child);
+            AZ_Assert(childEntity, "No child entity %s", child.ToString().c_str());
+
+            const auto* editorSpawnPoint = childEntity->FindComponent<ROS2SpawnPointEditorComponent>();
+
+            if (editorSpawnPoint != nullptr)
+            {
+                result.insert(editorSpawnPoint->GetInfo());
+            }
+        }
+
+        // setting name of spawn point component "default" in a child entity will have no effect since it is overwritten here with the
+        // default spawn pose of spawner
+        result["default"] = SpawnPointInfo{ "Default spawn pose defined in the Editor", m_controller.m_config.m_defaultSpawnPose };
+        return result;
+    }
+
+    const AZ::Transform& ROS2SpawnerEditorComponent::GetDefaultSpawnPose() const
+    {
+        return m_controller.m_config.m_defaultSpawnPose;
+    }
+
+    AZStd::unordered_map<AZStd::string, SpawnPointInfo> ROS2SpawnerEditorComponent::GetAllSpawnPointInfos() const
+    {
+        return GetSpawnPoints();
+    }
+
+    bool ROS2SpawnerEditorComponent::ShouldActivateController() const
+    {
+        return false;
+    }
+
+    void ROS2SpawnerEditorComponent::Activate()
+    {
+        ROS2SpawnerEditorComponentBase::Activate();
+        ROS2SpawnerComponentConfig config = m_controller.GetConfiguration();
+        config.m_editorEntityId = GetEntityId();
+        m_controller.SetConfiguration(config);
+        SpawnerRequestsBus::Handler::BusConnect(GetEntityId());
```

Comment:
No need to call `GetEntityId()` twice
```suggestion
        SpawnerRequestsBus::Handler::BusConnect(config.m_editorEntityId);
```